Add a new requirement to org membership - updated gitdm! (#7725)
kubernetes/community: community-membership.md

### Requirements

- Enabled [two-factor authentication] on their GitHub account

> - Updated their affiliation and email in [CNCF gitdm]

Reviewer: ```suggestion
- Updated their affiliation (can be "independent") and email in [CNCF gitdm]
```

Also, what if they choose not to divulge an email address through the noreply feature? I know people who choose not to give that out or even post it publicly in a modified form due to stalking concerns.

### Requirements

- Enabled [two-factor authentication] on their GitHub account

> - Updated their GitHub username, company affiliation and email in [CNCF gitdm]. If you are
>   not affiliated with a company please mark yourself as "Independent".

Reviewer 2: ```suggestion
- Ensure GitHub username, company affiliation and email in [CNCF gitdm] are
  up to date. If you are not affiliated with a company please mark yourself as
  "Independent". 
    - gitdm is primarily used by [devstats] to track contributions from the
      many companies involved in the ecosystem. Kubernetes also uses it to
      ensure org membership sponsors are from different member companies.
- Ensure affiliation is up to date in [openprofile.dev]. 
  - openprofile.dev will replace gitdm in the future to track affiliation.
```

Wanted to provide a little context as to why affiliation is important and call out that openprofile.dev will eventually be the new gitdm (even if it winds up being a few years away)

[continuously active]: #inactive-members
[sig-governance-subproject-lead]: /committee-steering/governance/sig-governance.md#subproject-lead
[sig-governance-subproject-owner]: /committee-steering/governance/sig-governance.md#subproject-owner

> [CNCF gitdm]: https://github.com/cncf/gitdm

Reviewer 2: ```suggestion
[CNCF gitdm]: https://github.com/cncf/gitdm
[devstats]: https://k8s.devstats.cncf.io/
[openprofile.dev] https://openprofile.dev/edit/profile
```
